Fix EITMap colormap loading and improve instrument detection

mgjeon opened the pull request:
Fixes: https://github.com/sunpy/sunpy/issues/8069

### Description
- Modify `self.plot_settings['cmap']` to load the colormap even when `OBSRVTRY` is not `"SOHO"`.
- Modify `is_datasource_for` to detect EIT data by checking if `TELESCOP` is `"Extreme-ultraviolet Imaging Telescope (EIT)"` even when `INSTRUME` is missing.

### Minimal Example
```python
# Download SOHO EIT 171 L1 FITS file
from urllib.request import urlretrieve
url = "https://umbra.nascom.nasa.gov/pub/eit/l1/2007/06/01/SOHO_EIT_171_20070601T120013_L1.fits"
urlretrieve(url, "SOHO_EIT_171_20070601T120013_L1.fits")

# Load the FITS file
from sunpy.map import Map
smap = Map("SOHO_EIT_171_20070601T120013_L1.fits")

# Can load cmap 'sohoeit171'
smap.peek()

# Can load EITMap
print(type(smap))  # 'sunpy.map.sources.soho.EITMap'
```
![output](https://github.com/user-attachments/assets/aa249660-da89-4132-94d7-8bce485a535e)

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -38,7 +38,7 @@ def __init__(self, data, header, **kwargs):
 
         self._nickname = self.detector
-        self.plot_settings['cmap'] = self._get_cmap_name()
+        self.plot_settings['cmap'] = f"sohoeit{str(int(self.wavelength.to('angstrom').value))}"
```
What does `self._get_cmap_name()` return for the level 1 file?

mgjeon replied:
It returns the string `"solar and heliospheric observatory (soho)eit171"`.

nabobalis commented:
Thanks for the PR @mgjeon, can you add a bugfix changelog file please following: https://github.com/sunpy/sunpy/blob/main/changelog/README.rst

mgjeon commented:
Thanks for the kind feedback, @nabobalis!

I’ve added the SOHO/EIT level 1 FITS headers (171, 195, 284, 304) to the `data/test/EIT_header` folder and included a cmap check test function named `test_l1_eit_map` in `test_eit_source.py`. I also added the changelog file. 

Please let me know if this is on the right track or if any adjustments are needed.

nabobalis commented on changelog/8070.bugfix.rst:
```diff
@@ -0,0 +1 @@
+Fixed a bug where `~sunpy.map.sources.EITMap` and the correct colormaps (e.g., `sohoeit171`) failed to load for SOHO/EIT level 1 FITS files from SDAC.
```
```suggestion
Fixed a bug where `~sunpy.map.sources.EITMap` and the correct colormaps (e.g., ``sohoeit171``) failed to load for SOHO/EIT level 1 FITS files from SDAC.
```

nabobalis commented on sunpy/map/sources/tests/test_eit_source.py:
```diff
@@ -76,3 +76,13 @@ def test_wcs(eit_map):
 def test_old_eit_date():
     eit_map = get_dummy_map_from_header(get_test_filepath("seit_00171_fd_19961211_1900.header"))
     assert eit_map.date.value == '1996-12-11T19:00:14.254'
+
```
```suggestion


```

nabobalis commented:
pre-commit.ci autofix

nabobalis commented:
Thank you @mgjeon
